### PR TITLE
Use u8string for custom paths

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/OpenRCT2.cpp [hunk 1]
```diff
@@ -11,10 +11,10 @@
...
 StartupAction gOpenRCT2StartupAction = StartupAction::Title;
 utf8 gOpenRCT2StartupActionPath[512] = { 0 };
-utf8 gCustomUserDataPath[MAX_PATH] = { 0 };
-utf8 gCustomOpenRCT2DataPath[MAX_PATH] = { 0 };
-utf8 gCustomRCT1DataPath[MAX_PATH] = { 0 };
-utf8 gCustomRCT2DataPath[MAX_PATH] = { 0 };
+u8string gCustomUserDataPath = {};
+u8string gCustomOpenRCT2DataPath = {};
+u8string gCustomRCT1DataPath = {};
+u8string gCustomRCT2DataPath = {};
```
Comment (tupaschoal) [inline src/openrct2/OpenRCT2.cpp:17]: The initializers are technically not needed, right? A string is well-formed without having to default initialize it.
```suggestion
u8string gCustomUserDataPath;
u8string gCustomOpenRCT2DataPath;
u8string gCustomRCT1DataPath;
u8string gCustomRCT2DataPath;
```

File: src/openrct2/platform/Platform.Linux.cpp [hunk 2]
```diff
@@ -89,7 +89,7 @@ namespace Platform
     std::string GetInstallPath()
     {
         // 1. Try command line argument
-        auto path = std::string(gCustomOpenRCT2DataPath);
+        auto path = gCustomOpenRCT2DataPath;
```
Comment (tupaschoal) [inline src/openrct2/platform/Platform.Linux.cpp:92]:
```suggestion
        const auto& path = gCustomOpenRCT2DataPath;
```
`const &` to avoid a copy here?

Comment (Broxzier) [reply src/openrct2/platform/Platform.Linux.cpp:92]: Now that changing to `std::string` is not necessary anymore, this entire variable can be ditched and `gCustomOpenRCT2DataPath` can be passed directly instead.

File: src/openrct2/OpenRCT2.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/OpenRCT2.cpp:17]: I’m not sure and I wouldn’t want to risk it :sweat_smile:

Comment (Broxzier) [reply src/openrct2/OpenRCT2.cpp:17]: Tulio is correct, `std::string` is in a valid state when initialized without specifying any constructor.

**Pull request merged**